src/query/plan/rule_based_planner.cpp

```diff
@@ -341,14 +345,17 @@ class ReturnBodyContext : public HierarchicalTreeVisitor {
     bool has_aggr = aggr1 || aggr2;                                                          \
     if (has_aggr && !(aggr1 && aggr2)) {                                                     \
       /* Group by the expression which does not contain aggregation. */                      \
-      /* Possible optimization is to ignore constant value expressions */                    \
-      group_by_.emplace_back(aggr1 ? op.expression2_ : op.expression1_);                     \
+      if (aggr1 && !IsConstantLiteral(op.expression2_)) {                                    \
+        group_by_.emplace_back(op.expression2_);                                             \
+      }                                                                                      \
+      if (aggr2 && !IsConstantLiteral(op.expression1_)) {                                    \
+        group_by_.emplace_back(op.expression1_);                                             \
+      }                                                                                      \
```

Comment:
Is it possible that happens that aggr1 is not aggregation, but aggr2 is, but op.expression_1 is IsConstantLiteral and then we don't add group by, but we emplace in has_aggregation_ that there is aggregation?

For example 

```
MATCH (n) 
RETURN 1 + sum(n.prop) AS prop_sum;
```

Reply:
Good catch, fixed

Reply:
Yes, it’s possible. In that case we have an aggregation on one side. Normally this means the planner has to group aggregations by the other side’s values, but since the other side is a constant (i.e. just one possibility), there’s no need to group by. This is as intended.